### PR TITLE
feat: Add Workload entities with straightforward facets

### PR DESCRIPTION
**Pull request by obs-gh-enricogiorio**
Add a variety of facets for the following new entities:

- Job
- CronJob
- StatefulSet
- DaemonSet

### CODE REVIEW

File: charts/agent/templates/_deployment-cluster-events-config.tpl [hunk 1]
```diff
@@ -186,6 +186,67 @@ processors:
...
+          - set(attributes["observe_transform"]["facets"]["updatedNumberScheduled"], body["status"]["updatedNumberScheduled"])
+          - set(attributes["observe_transform"]["facets"]["numberAvailable"], body["status"]["numberAvailable"])
+          - set(attributes["observe_transform"]["facets"]["numberUnavailable"], body["status"]["numberUnavailable"])
+          - set(attributes["observe_transform"]["facets"]["numberMisscheduled"], body["status"]["numberMisscheduled"])
+          # spec
+          - set(attributes["observe_transform"]["facets"]["updateStrategy"], body["spec"]["updateStrategy"]["type"])
+          - set(attributes["observe_transform"]["facets"]["maxUnavailable"], body["spec"]["updateStrategy"]["rollingUpdate"]["maxUnavailable"])
+          - set(attributes["observe_transform"]["facets"]["maxUnavailable"], 0) where attributes["observe_transform"]["facets"]["maxUnavailable"] == nil
+          - set(attributes["observe_transform"]["facets"]["maxSurge"], body["spec"]["updateStrategy"]["rollingUpdate"]["maxSurge"])
+          - set(attributes["observe_transform"]["facets"]["maxSurge"], 0) where attributes["observe_transform"]["facets"]["maxSurge"] == nil
```
Comment (xueweiz) [inline charts/agent/templates/_deployment-cluster-events-config.tpl:234]: curious, why do you set it to 0 when the field doesn't exist?
(i.e. is 0 better than nil? if so, under what situation, and why?

File: charts/agent/templates/_deployment-cluster-events-config.tpl [hunk 2]
```diff
@@ -186,6 +186,67 @@ processors:
...
+        conditions:
+          - body["kind"] == "StatefulSet"
+        statements:
+          # status
+          - set(attributes["observe_transform"]["facets"]["currentReplicas"], body["status"]["currentReplicas"])
+          - set(attributes["observe_transform"]["facets"]["readyReplicas"], body["status"]["readyReplicas"])
+          # spec
+          - set(attributes["observe_transform"]["facets"]["service"], body["spec"]["serviceName"])
+          - set(attributes["observe_transform"]["facets"]["podManagementPolicy"], body["spec"]["podManagementPolicy"])
+          - set(attributes["observe_transform"]["facets"]["replicas"], body["spec"]["replicas"])
```
Comment (xueweiz) [inline charts/agent/templates/_deployment-cluster-events-config.tpl:246]: desiredReplicas?
(that's what we call it for Deployment

File: charts/agent/templates/_deployment-cluster-events-config.tpl [hunk 3]
```diff
@@ -186,6 +186,67 @@ processors:
...
+          - set(attributes["observe_transform"]["facets"]["updateStrategy"], body["spec"]["updateStrategy"]["type"])
+          - set(attributes["observe_transform"]["facets"]["partition"], body["spec"]["updateStrategy"]["rollingUpdate"]["partition"])
+          - set(attributes["observe_transform"]["facets"]["partition"], 0) where attributes["observe_transform"]["facets"]["partition"] == nil
```
Comment (xueweiz) [inline charts/agent/templates/_deployment-cluster-events-config.tpl:249]: same as above, is 0 better than nil? For what use case?

File: charts/agent/templates/_deployment-cluster-events-config.tpl [hunk 4]
```diff
@@ -186,6 +186,67 @@ processors:
...
+          - set(attributes["observe_transform"]["facets"]["activePods"], body["status"]["active"])
+          - set(attributes["observe_transform"]["facets"]["failedPods"], body["status"]["falied"])
+          - set(attributes["observe_transform"]["facets"]["succeededPods"], body["status"]["succeeded"])
+          - set(attributes["observe_transform"]["facets"]["readyPods"], body["status"]["ready"])
+          # spec
+          - set(attributes["observe_transform"]["facets"]["completions"], body["spec"]["completions"])
+          - set(attributes["observe_transform"]["facets"]["parallelism"], body["spec"]["parallelism"])
+          - set(attributes["observe_transform"]["facets"]["activeDeadlineSeconds"], body["spec"]["activeDeadlineSeconds"])
+          - set(attributes["observe_transform"]["facets"]["backoffLimit"], body["spec"]["backoffLimit"])
+      # Cron Job
```
Comment (xueweiz) [inline charts/agent/templates/_deployment-cluster-events-config.tpl:205]: tiny nit: maybe just CronJob, DaemonSet

File: charts/agent/templates/_deployment-cluster-events-config.tpl (hunk 1 above)
Comment (obs-gh-enricogiorio) [reply charts/agent/templates/_deployment-cluster-events-config.tpl:234]: I had it nil, but then looking at our code we are already doing this thing for readyReplicas (Deployment and ReplicaSet). I think `nil` is better tho. I will revert this and remove the set to 0.

**Review by xueweiz — APPROVED** [review]

**Pull request merged**